src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1080,8 +1080,7 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMProposeSet> const& m)
 {
     protocol::TMProposeSet& set = *m;
 
-    if (overlay_.setup().expire &&
-            set.has_hops() && ! slot_->cluster())
+    if (set.has_hops() && ! slot_->cluster())
```

Comment:
Scott: the `expire` configuration only applies to validators: if it is `true` then the validator sets the hop count field in certain messages.

If present, the field is then incremented on every "hop" of its journey (except that intercluster hops don't count) and will be dropped once the hop increases beyond a certain limit.

The issue here was that we were incorrectly checking whether `expire` was set when we were _receiving_ a message and only if it was, would we then attempt to adjust the hop count, if present. 

The correct behavior is to always increment the hop count when the field is present, unless dealing with an inter-cluster hop.